### PR TITLE
Sections

### PR DESCRIPTION
**Pull request by BohdanBusko**
**STATUS**
READY

**DESCRIPTION**
add a column section(type boolean) to table tasks
update views for task and workspaces

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

### CODE REVIEW

File: spec/models/task_spec.rb [hunk 1]
```diff
@@ -23,30 +24,40 @@
...
+  let!(:task2) { create(:task, project_id: project.id) }
+  let!(:task3) { create(:task, project_id: project.id) }
+
+  context 'test before_save :section' do
+    let!(:task5) { create(:task, project_id: project.id) }
+    let!(:task6) { create(:task, title: 'Section:',  project_id: project.id) }
+    let!(:task7) { create(:task, title: 'Test : text') }
+
+    it 'should set status section as false' do
+      expect(task5.section).to eq false
```
Comment (romaslmd) [inline spec/models/task_spec.rb]: nitpick:
* `expect(task5).not_to be_section`
* `expect(task5.section?).to be_falsey`
* `expect(task5.section).to be_falsey`

File: app/views/account/projects/show.html.slim [hunk 2]
```diff
@@ -3,21 +3,21 @@
...
-         = link_to task.title, account_project_task_path(@project, task)
-       .level-item
-         = link_to "Up", move_account_project_task_path(@project, task,
-           task: { row_order_position: :up}), method: :put, class: "button is-small left"
-       .level-item
-         = link_to "Down", move_account_project_task_path(@project, task,
-           task: { row_order_position: :down }), method: :put, class: "button is-small left"
+        .level-item
+          = link_to task.title, account_project_task_path(@project, task),
+            task.section? ? { class: 'title is-4 is-italic' } : nil
```
Comment (romaslmd) [inline app/views/account/projects/show.html.slim]: fyi
`= link_to task.title, account_project_task_path(@project, task), class: ('title is-4 is-italic' if task.section?)`

File: spec/models/task_spec.rb [hunk 3]
```diff
@@ -23,30 +24,40 @@
...
 
 RSpec.describe Task, type: :model do
-  context 'factory tests' do
-    subject { build(:task) }
-    it { is_expected.to be_valid }
-  end
+  let!(:user) { create(:user) }
+  let!(:workspace) { create(:workspace, user_id: user.id) }
+  let!(:project) { create(:project, workspace_id: workspace.id) }
+  let!(:task1) { create(:task, project_id: project.id) }
```
Comment (romaslmd) [inline spec/models/task_spec.rb]: fyi: `create(:project, workspace: workspace)` to get rid of extra querying on validations

File: app/models/task.rb [hunk 4]
```diff
@@ -30,6 +31,11 @@ class Task < ApplicationRecord
   scope :row_order_asc, -> { order(row_order: :asc) }
 
   validates :title, length: { maximum: 250 }, presence: true
-  validates :description, length: { maximum: 250 }, presence: true
+  validates :description, length: { maximum: 250 }
 
+  before_save :section
+
+  def section
+    self.section = /[:]$/.match?(self.title.strip) ? true : false
```
Comment (romaslmd) [inline app/models/task.rb]: `self.section = self.title.strip.ends_with?(':')` would be much better

Comment (romaslmd) [reply app/models/task.rb]: fyi: https://stackoverflow.com/questions/577653/difference-between-a-z-and-in-ruby-regular-expressions

File: app/views/account/projects/show.html.slim (hunk 2 above)
Comment (igorkasyanchuk) [reply app/views/account/projects/show.html.slim]: or = link_to task.title, account_project_task_path(@project, task), class: task_class(task)

helper:

def task_class(task)
  'title is-4 is-italic' if task.section?
end

File: db/migrate/20181025132424_add_section_to_task.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddSectionToTask < ActiveRecord::Migration[5.2]
+  def change
+    add_column :tasks, :section, :boolean
```
Comment (igorkasyanchuk) [inline db/migrate/20181025132424_add_section_to_task.rb]: default: false

File: app/assets/stylesheets/bulma_ext.scss [hunk 6]
```diff
@@ -8,3 +8,26 @@
     }
   }
 }
+
+.workspaces {
+  .projects {
+    margin-top: 10px;
+  }
+
+  .lavel {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/bulma_ext.scss]: remove

File: app/controllers/account/tasks_controller.rb [hunk 7]
```diff
@@ -6,11 +6,11 @@ def show
 
   def new
     @project = parent
-    @task = @project.tasks.build
+    @task = @project.tasks.build(section: params[:section])
   end
```
Comment (romaslmd) [inline app/controllers/account/tasks_controller.rb:10]: unpermitted parameter

File: spec/controllers/account/tasks_controller_spec.rb [hunk 8]
```diff
@@ -3,20 +3,35 @@
...
-      sign_in user
-      user.update(projects: [project])
+  let!(:user) { create(:user) }
+  let!(:workspace) { create(:workspace, user: user) }
+  let!(:project) { create(:project, workspace: workspace) }
+  let!(:task) { create_list(:task, 5, project: project) }
+
+  before do
+    sign_in user
+    user.update(projects: [project])
```
Comment (romaslmd) [inline spec/controllers/account/tasks_controller_spec.rb]: fyi
```
let!(:user) { create(:user, projects: [project] } 
```

Comment (romaslmd) [reply spec/controllers/account/tasks_controller_spec.rb]: why u have such complicated assoc chain?
shouldn't it be like `user :has_many projects through: :workplace`?

File: spec/controllers/account/tasks_controller_spec.rb [hunk 9]
```diff
@@ -3,20 +3,35 @@
...
+  end
+
+  after do
+    sign_out user
+  end
+
+  context 'GET /projects/:project_id/task/:id' do
+    it 'should show task page' do
+      get :show, params: { project_id: project.id, id: task.first.id }
+      expect(response).to be_successful
```
Comment (romaslmd) [inline spec/controllers/account/tasks_controller_spec.rb:18]: be_successful? does it work?

Comment (BohdanBusko) [reply spec/controllers/account/tasks_controller_spec.rb:18]: DEPRECATION WARNING: The success? predicate is deprecated and will be removed in Rails 6.0. Please use successful? as provided by Rack::Response::Helpers. (called from block (3 levels) in <top (required)> at /home/bogdan/Documents/Workspace/asana/spec/controllers/account/tasks_controller_spec.rb:30

File: app/views/account/workspaces/show.html.slim [hunk 10]
```diff
@@ -1,3 +1,22 @@
-- title @workspace.name
-.title
-  = @workspace.name
\ No newline at end of file
+.container
+  .level
+    .level-left
+      .title
```
Comment (igorkasyanchuk) [inline app/views/account/workspaces/show.html.slim:4]: don't you need to add "level-item" here http://prntscr.com/lbheqx ?
as below, looks like in bulma it's required

File: spec/controllers/account/tasks_controller_spec.rb [hunk 11]
```diff
@@ -3,20 +3,34 @@
...
+  let!(:workspace) { create(:workspace, user: user) }
+  let!(:project) { create(:project, workspace: workspace, users: [user]) }
+  let!(:task) { create_list(:task, 5, project: project) }
+
+  before do
+    sign_in user
+  end
+
+  after do
+    sign_out user
```
Comment (igorkasyanchuk) [inline spec/controllers/account/tasks_controller_spec.rb]: why you need to do sign_out?

File: spec/controllers/account/workspaces_controller_spec.rb [hunk 12]
```diff
@@ -28,6 +24,20 @@
     end
   end
 
+  context 'GET /worspaces/:id' do
+    it 'should show page with workspace' do
+      get :show, params: { id: workspace }
```
Comment (romaslmd) [inline spec/controllers/account/workspaces_controller_spec.rb]: `workspace.to_param`?

**Pull request merged**